kip-0010.md

```diff
@@ -0,0 +1,198 @@
+
+### 5. Activation
+The features introduced in this KIP are activated based on DAA score:
+1. Prior to activation:
+   - New opcodes are treated as invalid
+   - Arithmetic operations remain limited to 4 bytes
+2. After activation:
+   - All new opcodes become available
+   - 8-byte arithmetic support is enabled
+   - Existing scripts continue to function as before
+
+### 6. Reserved Opcodes
+The following opcodes are reserved for future expansion:
+- OpTxVersion (0xb2)
+- OpTxLockTime (0xb5)
+- OpTxSubnetId (0xb6)
+- OpTxGas (0xb7)
+- OpTxPayload (0xb8)
+- OpOutpointTxId (0xba)
+- OpOutpointIndex (0xbb)
+- OpTxInputScriptSig (0xbc)
+- OpTxInputSeq (0xbd)
+- OpTxInputBlockDaaScore (0xc0)
+- OpTxInputIsCoinbase (0xc1)
+
+## Rationale
+
+The enhanced opcodes and arithmetic capabilities address several key requirements:
+
+1. Transaction Introspection: Scripts can now examine and validate transaction properties directly, enabling complex conditional logic
+2. Larger Number Support: 8-byte integers allow for precise calculations with large values, essential for financial operations
+3. Future Extensibility: Reserved opcodes provide a clear path for future enhancements
+4. Backward Compatibility: The activation mechanism ensures a smooth network upgrade
+
+These opcodes are designed to work within the existing P2SH framework, maintaining compatibility with current address types while significantly expanding the possibilities for script design.
+
+## Backwards Compatibility
+
+This proposal requires a hard fork, as it introduces new opcodes to the scripting language. Older software will require an update to support these new features. Existing scripts and addresses remain valid, but cannot use the new functionality without being updated.
+
+## Reference Implementation
+
+A reference implementation of the new opcodes and example usage can be found in the following pull request to the rusty-kaspa repository:
+
+[https://github.com/kaspanet/rusty-kaspa/pull/487](https://github.com/kaspanet/rusty-kaspa/pull/487)
+
+Additional examples demonstrating the use of these opcodes can be found in:
+
+[https://github.com/biryukovmaxim/rusty-kaspa/blob/kip-10-mutual-tx/crypto/txscript/examples/kip-10.rs](https://github.com/biryukovmaxim/rusty-kaspa/blob/kip-10-mutual-tx/crypto/txscript/examples/kip-10.rs)
+
+## Example Usage
+
+To illustrate the practical application of the new opcodes, we present three scenarios: a threshold scenario, a shared secret scenario, and a mining pool payment scenario.
+
+### 1. Threshold Scenario
+
+This scenario demonstrates a script that allows for two types of spending conditions:
+
+1. Owner spending: The owner can spend the UTXO by providing a valid signature.
+2. Borrower spending: Anyone can spend the UTXO if they create an output with a value greater than the input by a specified threshold amount, sent to the same script.
```

Comment:
This should contain an explanation about why this is helpful for micro payments after KIP 9

Reply:
https://github.com/biryukovmaxim/kips/blob/4e33f0f76410bb7895ab538a47178a865b6f85ce/kip-0010.md?plain=1#L113-L120
ive tried to explain it there